docs/rules/spec-strict-refs.md

```diff
@@ -0,0 +1,99 @@
+# spec-strict-refs
+
+Restricts the usage of the `$ref` keyword.
```

Comment:
Let's instead say something like "Check that $ref is only used in the locations permitted by the OpenAPI specification." I think it helps to give a bit more context about why you'd want that.